### DISCUSSION TITLE
docs/guide/type-safety

### DISCUSSION THREAD

**Discussion by giscus** [mikro-orm/mikro-orm#5237] [Comments]
# docs/guide/type-safety

Entity relations are mapped to entity references - instances of the entity that have at least the primary key available. This reference is stored in Identity Map, so you will get the same object reference when fetching the same document from database.

https://mikro-orm.io/docs/guide/type-safety

**Comment by andrewzenkov** [comment]
The work with type protection when we populate entities is just fantastic. 

Operator $ helps to make it more elegant. 
```javascript
const userName = user.profile.$.name
```
